### DISCUSSION TITLE
The "whisper" for the individual time codes.

### DISCUSSION THREAD

**Discussion by frezer2011** [SubtitleEdit/subtitleedit#6976] [Ideas]
Hello! Make the speech recognition "whisper" not the entire file, but a separate piece of time. that would be possible to run the individual time codes by other more heavy models.

**Comment by darnn** [comment]
I'm not entirely sure what you mean, but I assume it's one of two things:
You can run Whisper on a specific subtitle by right-clicking it and then going to selected lines->audio->audio to text (Whisper).
Conversely, if you want to just extract a specific piece of audio to use Whisper on, simply create a subtitle with the start and end timecodes of your audio (for example, if you want just the minute between 01:00 and 02:00, your subtitle should start at 01:00 and end at 02:00), right-click it, and go to selected lines->audio->extract audio. This will let you save the audio between those timecodes, and you can use it with other more heavy models or whatever  it is.
I assume you might also mean that you want SE to determine the timecodes where audio exists, and then use Whisper on just those timecodes. There's no truly good way of doing this right now, but I suppose your best bet would be to use Whisper on an entire file with the tiny model, and then use the timecodes from that. Note however that they won't necessarily be accurate, so this can end up doing more harm than good. The best results I've gotten so far in terms of timecode accuracy were in WhisperX, if you can figure out how to install and run it.

**Reply by frezer2011** [reply]
Thank you so much! That's just what I need. I was looking for this function on the timeline, not in the line.  Unfortunately, I have not yet figured out how to install WhisperX. The SE asks for the path to WhisperX.exe. I don't understand where to get it yet. I haven't found the manual yet either. Thanks for the reply. It helped me a lot.
